airbyte/_processors/sql/snowflakecortex.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+"""A Snowflake vector store implementation of the SQL processor."""
+
+from __future__ import annotations
+
+from textwrap import dedent, indent
+from typing import TYPE_CHECKING
+
+import sqlalchemy
+from overrides import overrides
+
+from airbyte_cdk.models import AirbyteMessage, ConfiguredAirbyteCatalog
+from airbyte._processors.sql.snowflake import SnowflakeSqlProcessor, SnowflakeTypeConverter
+from airbyte._processors.base import RecordProcessor
+
+from typing import TYPE_CHECKING, Any, cast
+from airbyte_cdk.models import (
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStateMessage,
+    AirbyteStateType,
+    AirbyteStreamState,
+    ConfiguredAirbyteCatalog,
+    ConfiguredAirbyteStream,
+    Type,
+)
+from sqlalchemy.engine import Connection, Engine
+from airbyte._processors.file.base import FileWriterBase
+from airbyte.strategies import WriteStrategy
+from airbyte import exceptions as exc
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+    from pathlib import Path
+    from airbyte.caches.base import CacheBase
+    from sqlalchemy.engine import Connection
+
+
+class SnowflakeCortexTypeConverter(SnowflakeTypeConverter):
+    """A class to convert array type into vector."""
+
+    def __init__(
+        self,
+        conversion_map: dict | None = None,
+        *,
+        vector_length: int,
+    ) -> None:
+        self.vector_length = vector_length
+        super().__init__(conversion_map)
+
+    @overrides
+    def to_sql_type(
+        self,
+        json_schema_property_def: dict[str, str | dict | list],
+    ) -> sqlalchemy.types.TypeEngine:
+        """Convert a value to a SQL type."""
+        sql_type = super().to_sql_type(json_schema_property_def)
+        if isinstance(sql_type, sqlalchemy.types.ARRAY):
+            return f"Vector(Float, {self.vector_length})"
+
+        return sql_type
+
+
+class SnowflakeCortexSqlProcessor(SnowflakeSqlProcessor):
+    """A Snowflake implementation for use with Cortex functions."""
+
+    type_converter_class: SnowflakeCortexTypeConverter
+
+    def __init__(
```

Comment:
overriding the constructor since there was not way to pass the vector_length variable to TypeConverter otherwise.

Reply:
Makes sense. When we're ready to merge, I can optionally do another pass to see if there's any way to reduce duplication (maybe call `super()`?). But as it is, I think this is okay.

Reply:
super() does a few things we actually want to override, that's why doing the entire initialization here.